glib-macros/src/properties.rs

```diff
@@ -0,0 +1,527 @@
+            );
+            quote!(glib::ParamFlags::empty() #(| glib::ParamFlags::#flags_iter)*)
+        };
+
+        let builder_call = builder
+            .as_ref()
+            .cloned()
+            .map(|(mut required_params, opts)| {
+                let name_expr = syn::ExprLit {
+                    attrs: vec![],
+                    lit: syn::Lit::Str(name.to_owned()),
+                };
+                required_params.insert(0, name_expr.into());
+                let required_params = required_params.iter();
+
+                quote!((#(#required_params,)*)#opts)
+            })
+            .unwrap_or(quote!((#name)));
+
+        let build_nick = nick.as_ref().map(|x| quote!(.nick(#x)));
+        let build_blurb = blurb.as_ref().map(|x| quote!(.blurb(#x)));
+        quote! {
+            <<#ty as glib::Property>::ParamSpec>
+                ::builder #builder_call
+                #build_nick
+                #build_blurb
+                .flags(#flags)
+                .build()
+        }
+    });
+    quote!(
+        fn derived_properties() -> &'static [glib::ParamSpec] {
+            use glib::once_cell::sync::Lazy;
+            static PROPERTIES: Lazy<[glib::ParamSpec; #n_props]> = Lazy::new(|| [
+                #(#properties_build_phase,)*
+            ]);
+            PROPERTIES.as_ref()
+        }
+    )
+}
+fn expand_property_fn(props: &[PropDesc]) -> TokenStream2 {
+    let match_branch_get = props.iter().flat_map(|p| {
+        let PropDesc {
+            name,
+            field_ident,
+            member,
+            get,
+            ..
+        } = p;
+        get.as_ref().map(|get| match (member, get) {
+            (_, MaybeCustomFn::Custom(expr)) => quote!(
+                #name => Ok((#expr)(&self).to_value())
+            ),
+            (None, MaybeCustomFn::Default) => quote!(
+                #name => Ok(self.#field_ident.get(|v| v.to_value()))
+            ),
+            (Some(member), MaybeCustomFn::Default) => quote!(
+                #name => Ok(self.#field_ident.get(|v| v.#member.to_value()))
+            ),
+        })
+    });
+    quote!(
+        fn derived_property<'a>(&self, _obj: &Self::Type, _id: usize, pspec: &'a glib::ParamSpec) -> Result<glib::Value, glib::subclass::object::MissingPropertyHandler<'a>> {
+            match pspec.name() {
+                #(#match_branch_get,)*
+                p => Err(pspec.into())
+            }
+        }
+    )
+}
+fn expand_set_property_fn(props: &[PropDesc]) -> TokenStream2 {
+    let match_branch_set = props.iter().flat_map(|p| {
+        let PropDesc {
+            name,
+            field_ident,
+            member,
+            set,
+            ..
+        } = p;
+
+        let expect = quote!(.expect("Can't convert glib::value to property type"));
+        set.as_ref().map(|set| match (member, set) {
+            (_, MaybeCustomFn::Custom(expr)) => quote!(
+                #name => {
```

Comment:
Can we use an equality comparison on the `ParamSpec` here, instead of doing a string comparison? That would be faster.